bool enable
float32 frequency
float32 intensity
